From 4e5ff1f188dd5216f010c34023a631cc86ffb5d3 Mon Sep 17 00:00:00 2001
Message-Id: <patch-14.17-4e5ff1f188d-20210711T153247Z-avarab@gmail.com>
In-Reply-To: <cover-00.17-00000000000-20210711T153247Z-avarab@gmail.com>
References: <pull.1012.v5.git.git.1625684869.gitgitgadget@gmail.com>
	<cover-00.17-00000000000-20210711T153247Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Sun, 11 Jul 2021 17:02:53 +0200
Subject: [PATCH v6? 14/17] refs API: make refs_ref_exists() not set errno
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Move refs_ref_exists from the legacy refs_resolve_ref_unsafe() to the
new refs_resolve_ref_unsafe_with_errno(). I have read its callers and
determined that they don't care about errno being set, in particular:

    git grep -W -w -e refs_ref_exists -e ref_exists

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 refs.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/refs.c b/refs.c
index 45b51c0a532..0364b68f8fa 100644
--- a/refs.c
+++ b/refs.c
@@ -329,7 +329,10 @@ int read_ref(const char *refname, struct object_id *oid)
 
 int refs_ref_exists(struct ref_store *refs, const char *refname)
 {
-	return !!refs_resolve_ref_unsafe(refs, refname, RESOLVE_REF_READING, NULL, NULL);
+	int ignore_errno;
+	return !!refs_resolve_ref_unsafe_with_errno(refs, refname,
+						    RESOLVE_REF_READING, NULL,
+						    NULL, &ignore_errno);
 }
 
 int ref_exists(const char *refname)
-- 
2.32.0-dev

